galaxy_ng/app/api/ui/viewsets/user.py

```diff
@@ -0,0 +1,43 @@
+from rest_framework import mixins
+from rest_framework import permissions as drf_permissions
+
+from galaxy_ng.app.models import auth as auth_models
+from galaxy_ng.app.api import permissions
+from galaxy_ng.app.api.ui import serializers
+from galaxy_ng.app.api import base as api_base
+
+
+class UserViewSet(
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.CreateModelMixin,
+    mixins.UpdateModelMixin,
+    api_base.GenericViewSet,
+):
+    permission_classes = [
+        permissions.IsPartnerEngineer,
+        permissions.RestrictOnCloudDeployments]
+
+    def get_serializer_class(self):
+        return serializers.UserSerializer
+
+    def get_queryset(self):
+        return auth_models.User.objects.all()
+
+
+class CurrentUserViewSet(
+    api_base.GenericViewSet,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+):
+    serializer_class = serializers.CurrentUserSerializer
+    model = auth_models.User
+    permission_classes = [
+        drf_permissions.IsAuthenticated,
```

Comment:
This will override default permission classes set in project settings. This is probably not that is wanted.
Better override `get_permissions()` method and append permissions to the default ones.